[jit] fix bug in CompilationUnit::define

suo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#21886 [jit] fix bug in CompilationUnit::define**

We were re-ordering definitions to put `__init__` first, but not
reordering resolvers. In reality, resolvers are always the same so this
doesn't matter. I may make `define()` take a single resolver in a future
PR.

Either way, this change is good because it avoids a copy of the whole
AST during `define()` as well.

Differential Revision: [D15867647](https://our.internmc.facebook.com/intern/diff/D15867647)

suo committed:
[jit] fix bug in CompilationUnit::define

We were re-ordering definitions to put `__init__` first, but not
reordering resolvers. In reality, resolvers are always the same so this
doesn't matter. I may make `define()` take a single resolver in a future
PR.

Either way, this change is good because it avoids a copy of the whole
AST during `define()` as well.

zdevito reviewed (approved):
NGNT. Definitely want to see us merge resolver into a single argument. They are all the same and managing them is annoying.

suo committed:
Update on "[jit] fix bug in CompilationUnit::define"

[jit] fix bug in CompilationUnit::define

We were re-ordering definitions to put `__init__` first, but not
reordering resolvers. In reality, resolvers are always the same so this
doesn't matter. I may make `define()` take a single resolver in a future
PR.

Either way, this change is good because it avoids a copy of the whole
AST during `define()` as well.

gh-metadata: pytorch pytorch 21886 gh/suo/64/head

suo committed:
Update on "[jit] fix bug in CompilationUnit::define"

[jit] fix bug in CompilationUnit::define

We were re-ordering definitions to put `__init__` first, but not
reordering resolvers. In reality, resolvers are always the same so this
doesn't matter. I may make `define()` take a single resolver in a future
PR.

Either way, this change is good because it avoids a copy of the whole
AST during `define()` as well.

gh-metadata: pytorch pytorch 21886 gh/suo/64/head
